<section>
  <div class="jumbotron shadow-lg" id="bigBox">
    <img src="../../../assets/img/spongEffects_logo.png" width="12%" height="12%"
         style="float: right; padding-left: 20px">
    <h1 class="display-4" id="headline">
      Explore spongEffects on TCGA projects and custom expressions
    </h1>
    <hr class="my-4">
    <p></p>
    <mat-expansion-panel class="info-card" expanded="false">
      <mat-expansion-panel-header>
        <mat-panel-title><mat-icon>info</mat-icon> What is spongEffects?</mat-panel-title>
        <mat-panel-description>
          General information about spongEffects on TCGA and custom usage
        </mat-panel-description>
      </mat-expansion-panel-header>
      <div class="info-text-div">
        <p><a href="https://pubmed.ncbi.nlm.nih.gov/37084275/">spongEffects</a> can be used for downstream analysis of SPONGE ceRNA networks and is capable of tumor sub-type classification and biomarker discovery via a random forest machine learning approach.</p>
        <p>For more detailed information refer to the <a href="https://pubmed.ncbi.nlm.nih.gov/37084275/">publication</a>.</p>
        <p>We trained spongEffects prediction models on the TCGA pan-cancer dataset and nine other TCGA projects that feature sufficient cancer subtyping for classification.</p>
        <p>
          The <img src="{{tabs[0].icon}}" width="20px" height="12px" style="padding-right: 5px">{{tabs[0].viewValue}} tab below shows the spongEffects prediction results for a selected TCGA project.
          You can browse the hub-nodes that spongEffects models views as potential biomarkers and compare their gene/transcript expression.
        </p>
        <p>
          We can use the pretrained TCGA models to classify tumor (sub-)types and predict biomarkers on new independent input data that can be uploaded via the
          <img src="{{tabs[1].icon}}" width="20px" height="12px" style="padding-right: 5px">{{tabs[1].viewValue}} tab below.
        </p>
        <p>The model will predict a tumor type for every sample in the uploaded expression file individually.</p>
      </div>
    </mat-expansion-panel>
    <hr>
    <h2>Explore TCGA results or predict on custom data</h2>
    <p></p>
    <mat-sidenav-container class="drawer-container" hasBackdrop="false">
      <mat-sidenav mode="side" opened class="sidebar">
        <div class="sidebar-tabs">
          <button mat-button *ngFor="let tab of tabs"
                  (click)="setTab(tab)"
                  class="sidebar-tab">
            <img src="{{tab.icon}}" width="30%" height="30%" style="padding-right: 5px">{{tab.viewValue}}
          </button>
        </div>
      </mat-sidenav>
      <mat-sidenav-content class="main-content-div">
        <!-- explore card-->
        <div class="explore-content" [hidden]="selectedTab.value!=tabs[0].value">
          <mat-card class="cancer-select-card">
            <mat-card-header>
              <mat-card-title>Select a project:</mat-card-title>
              <mat-card-subtitle>
                Choose one of nine processed TCGA projects or pan-cancer
              </mat-card-subtitle>
              <div class="cancer-select">
                <div class="col-md select-field">
                  <mat-form-field class="cancer-select-input">
                    <mat-label>Choose a TCGA project:</mat-label>
                    <mat-select [(ngModel)]="selectedCancer">
                      <mat-option *ngFor="let cancer of cancers"
                                  (mouseover)="setPreviewCancer(cancer)"
                                  [value]="cancer"
                                  (click)="cancerSelected(cancer); accordion.closeAll()">
                        {{cancer.toString()}}
                      </mat-option>
                    </mat-select>
                  </mat-form-field>
                </div>
              </div>
              <div class="level-toggle">
                <mat-button-toggle-group
                  name="level"
                  matTooltip="Select either gene or transcript level">
                  <mat-button-toggle *ngFor="let l of levels"
                                     value="{{l}}"
                                     class="level-toggle-button"
                                     (click)="setLevel(l)"
                                     style="{{getLevelButtonStyle(l)}}">
                    {{l}}
                  </mat-button-toggle>
                </mat-button-toggle-group>
              </div>
            </mat-card-header>
            <mat-card-content class="cancer-select-box">
              <mat-grid-list cols="100" rowHeight="10px">
                <mat-grid-tile colspan="30" rowspan="50">
                  <div class="cancer-image">
                    <img src="{{getCancerImage()}}">
                  </div>
                </mat-grid-tile>
                <mat-grid-tile colspan="70" rowspan="20" class="text-tile">
                  <div class="cancer-info-div">
                    <p>{{getCancerInfoText().text}}</p>
                    <p>Publication: <a class="publication-link" href="{{getCancerInfoText().link}}">{{getCancerInfoText().link}}</a></p>
                  </div>
                </mat-grid-tile>
                <mat-grid-tile colspan="70" rowspan="30">
                  <div class="sample-distribution-pie" #sampleDistributionPie></div>
                </mat-grid-tile>
              </mat-grid-list>
            </mat-card-content>
          </mat-card>
          <hr>
          <mat-expansion-panel class="result-panel" expanded="true">
            <mat-expansion-panel-header>
              <mat-panel-title>Results</mat-panel-title>
              <mat-panel-description>
                View spongEffects predictions for selected project
              </mat-panel-description>
            </mat-expansion-panel-header>
            <mat-accordion #accordion="matAccordion" class="result-accordion" [multi]="true">
              <mat-expansion-panel>
                <mat-expansion-panel-header (click)="resetOverallAccPlot()">
                  <mat-panel-title>Model performance</mat-panel-title>
                  <mat-panel-description>
                    Accuracy of spongEffects modules against random modules
                  </mat-panel-description>
                </mat-expansion-panel-header>
                <mat-expansion-panel class="info-card-small" expanded="false">
                  <mat-expansion-panel-header>
                    <mat-panel-title><mat-icon>info</mat-icon>What does this mean?</mat-panel-title>
                  </mat-expansion-panel-header>
                  <div class="info-text-small-div">
                    <p>
                      The models performance is compared against a model that was trained on randomly selected SPONGE centralities.
                      The model that was trained on the spongEffects modules should outperform the random approach.
                      Dotted lines represent the overall accuracy for the test split of the data and solid lines show analogously show the training performance.
                      Each line start (circle) and end (diamond) mark the lower and upper balanced accuracy bounds of a specific model.
                    </p>
                  </div>
                </mat-expansion-panel>
                <div class="overall-acc-plot-div">
                  <div *ngIf="overallAccuracyLoading" class="loading-spinner">
                    <img src="../../../assets/img/spongEffects_logo.png" alt="spongEffects loading spinner">
                  </div>
                  <div #overallAccuracyPlot [hidden]="overallAccuracyLoading" id="overall-acc" class="overall-acc-plot"></div>
                </div>
              </mat-expansion-panel>
              <p></p>
              <mat-expansion-panel #selectionPanel
                                   (opened)="performanceSelectPanelIsOpen = true"
                                   (closed)="performanceSelectPanelIsOpen = false">
                <mat-expansion-panel-header (click)="togglePanel($event, selectionPanel)">
                  <mat-panel-title>Class performance</mat-panel-title>
                  <mat-panel-description>
                    Class specific performance
                  </mat-panel-description>
                  <mat-form-field class="measure-select-input">
                    <mat-select [(ngModel)]="performanceMeasure"
                                name="score-select"
                                (click)="cancelClick($event)"
                                (selectionChange)="rePlotModelClassPerformance()">
                      <mat-option *ngFor="let p of performanceMeasures"
                                  [value]="p">
                        {{p.viewValue}}
                      </mat-option>
                    </mat-select>
                  </mat-form-field>
                </mat-expansion-panel-header>
                <mat-expansion-panel class="info-card-small" expanded="false">
                  <mat-expansion-panel-header>
                    <mat-panel-title><mat-icon>info</mat-icon>What does this mean?</mat-panel-title>
                  </mat-expansion-panel-header>
                  <div class="info-text-small-div">
                    <p>This plot shows the performance of the individual class predictions of the model that was trained with the spongEffects central modules against a model with randomly selected modules. We can strengthen the confidence in the models prediction if it is capable of outperforming its random counterpart. Interpretation of predictions should be made with more caution if this it not the case.</p>
                  </div>
                </mat-expansion-panel>
                <div class="class-acc-plot-div">
                  <div *ngIf="classPerformanceLoading" class="loading-spinner">
                    <img src="../../../assets/img/spongEffects_logo.png" alt="spongEffects loading spinner">
                  </div>
                  <div #classModelPerformancePlot [hidden]="classPerformanceLoading" class="class-acc-plot"></div>
                </div>
              </mat-expansion-panel>
              <p></p>
              <mat-expansion-panel>
                <mat-expansion-panel-header (click)="resetEnrichmentScoreByClass()">
                  <mat-panel-title>Classification</mat-panel-title>
                  <mat-panel-description>
                    Classification through enrichment score distributions
                  </mat-panel-description>
                </mat-expansion-panel-header>
                <mat-expansion-panel class="info-card-small" expanded="false">
                  <mat-expansion-panel-header>
                    <mat-panel-title><mat-icon>info</mat-icon>What does this mean?</mat-panel-title>
                  </mat-expansion-panel-header>
                  <div class="info-text-small-div">
                    <p>The spongEffects enrichment score distributions can give an insight into the model capability to differentiate between the given classes.</p>
                  </div>
                </mat-expansion-panel>
                <div class="enrichmentPlotHolderDiv">
                  <div *ngIf="enrichmentScoreDensityLoading" class="loading-spinner">
                    <img src="../../../assets/img/spongEffects_logo.png" alt="spongEffects loading spinner">
                  </div>
                  <div #enrichmentScoresByClassPlot [hidden]="enrichmentScoreDensityLoading" class="enrichmentPlotHolder"></div>
                </div>
              </mat-expansion-panel>
              <p></p>
              <mat-expansion-panel>
                <mat-expansion-panel-header (click)="resetLollipop()">
                  <mat-panel-title>Top ceRNA centralities</mat-panel-title>
                  <mat-panel-description>
                    View predicted centralities with the highest decrease in Gini index.
                  </mat-panel-description>
                </mat-expansion-panel-header>
                <mat-expansion-panel class="info-card-small" expanded="false">
                  <mat-expansion-panel-header>
                    <mat-panel-title><mat-icon>info</mat-icon>What does this mean?</mat-panel-title>
                  </mat-expansion-panel-header>
                  <div class="info-text-small-div">
                    <p>The mean decrease in the Gini coefficient tells us how much each variable affects the consistency of nodes and leaves in a random forest. If the mean decrease accuracy or mean decrease Gini score is higher, it means the variable is more important in the model.</p>
                    <p>Module centralities that are at the top right of the plot below play a crucial role in distinguishing between the various cancer (sub-)types.</p>
                  </div>
                </mat-expansion-panel>
                <div class="lollipop-select-box">
                  <mat-grid-list cols="100" rowHeight="10px">
                    <mat-grid-tile colspan="100" rowspan="10">
                      Select how many top rated modules you would like to include in the expression analysis and press the 'Investigate expressions' button or manually select the modules by clicking on them in the plot.
                    </mat-grid-tile>
                    <mat-grid-tile colspan="70" rowspan="30">
                      <div *ngIf="lollipopLoading" class="loading-spinner">
                        <img src="../../../assets/img/spongEffects_logo.png" alt="spongEffects loading spinner">
                      </div>
                      <div class="top-centralities-plot-div">
                        <div #lollipopPlot [hidden]="lollipopLoading" class="top-centralities-plot"></div>
                      </div>
                    </mat-grid-tile>
                    <mat-grid-tile colspan="30" rowspan="5">
                      <div class="select-top-n">
                        <mat-form-field class="top-select">
                          <mat-label>Number of top rated centralities to include</mat-label>
                          <input class="col-xs-4 col-xs-offset-2" type="string"
                                 matInput id="markNSelect"
                                 [formControl]="markControl"
                                 (keyup)="resetLollipop()">
                          <mat-error>Please provide a number between 3 and 20</mat-error>
                        </mat-form-field>
                      </div>
                    </mat-grid-tile>
                    <mat-grid-tile colspan="30" rowspan="5">
                      <div class="select-top-n">
                        <mat-form-field class="top-select">
                          <mat-label>Number of top rated centralities to display</mat-label>
                          <input class="col-xs-4 col-xs-offset-2" type="string"
                                 matInput id="topNSelect"
                                 [formControl]="topControl"
                                 (keyup)="resetLollipop()">
                          <mat-error>Please provide a number between 3 and 20</mat-error>
                        </mat-form-field>
                      </div>
                    </mat-grid-tile>
                    <mat-grid-tile colspan="30" rowspan="5">
                      <button mat-button class="run-button"
                              (click)="clearSelection()">
                        Clear selection
                      </button>
                    </mat-grid-tile>
                    <mat-grid-tile colspan="30" rowspan="5">
                      <mat-checkbox class="check-box"
                                    color="primary"
                                    [(ngModel)]="includeModuleMembers">
                        Include module members
                      </mat-checkbox>
                    </mat-grid-tile>
                    <mat-grid-tile colspan="30" rowspan="5">
                      <button mat-button class="run-button"
                              (click)="exploreExpression()">
                        Investigate expressions
                      </button>
                    </mat-grid-tile>
                    <mat-grid-tile colspan="100" rowspan="34">
                      <div class="modules-table">
                        <table mat-table [dataSource]="dynamicModulesData" class="mat-elevation-z8">
                          <ng-container matColumnDef="ensemblID">
                            <th mat-header-cell *matHeaderCellDef>ID</th>
                            <td mat-cell *matCellDef="let el">{{el.ensemblID}}</td>
                          </ng-container>
                          <ng-container matColumnDef="symbol">
                            <th mat-header-cell *matHeaderCellDef>Gene symbol</th>
                            <td mat-cell *matCellDef="let el">
                              {{el.symbol }}
                              <a href="{{get_gene_card(el.symbol)}}" target="_blank">
                                <img src="../../../assets/img/gene_card_icon.png" style="width:5%;height:5%;" alt="gene-card">
                              </a>
                            </td>
                          </ng-container>
                          <ng-container matColumnDef="meanGiniDecrease">
                            <th mat-header-cell *matHeaderCellDef>Mean Gini Decrease</th>
                            <td mat-cell *matCellDef="let el">{{el.meanGiniDecrease}}</td>
                          </ng-container>
                          <ng-container matColumnDef="meanAccuracyDecrease">
                            <th mat-header-cell *matHeaderCellDef>Mean Accuracy Decrease</th>
                            <td mat-cell *matCellDef="let el">{{el.meanAccuracyDecrease}}</td>
                          </ng-container>
                          <ng-container matColumnDef="description">
                            <th mat-header-cell *matHeaderCellDef>Description</th>
                            <td mat-cell *matCellDef="let el">{{el.description}}</td>
                          </ng-container>

                          <tr mat-header-row *matHeaderRowDef="modulesTableColumns; sticky: true"></tr>
                          <tr mat-row *matRowDef="let row; columns: modulesTableColumns;"></tr>
                        </table>
                      </div>
                    </mat-grid-tile>
                    <mat-grid-tile colspan="50" rowspan="30">
                      <div class="module-expression-heatmap">
                        <div *ngIf="elementExpressionLoading" class="loading-spinner">
                          <img src="../../../assets/img/spongEffects_logo.png" alt="spongEffects loading spinner">
                        </div>
                        <div #moduleExpressionHeatmapDiv [hidden]="elementExpressionLoading"></div>
                      </div>
                    </mat-grid-tile>
                    <mat-grid-tile colspan="50" rowspan="30">
                      <div class="module-expression-heatmap">
                        <div *ngIf="elementExpressionLoading" class="loading-spinner">
                          <img src="../../../assets/img/spongEffects_logo.png" alt="spongEffects loading spinner">
                        </div>
                        <div #moduleMiRnaExpressionDiv [hidden]="elementExpressionLoading"></div>
                      </div>
                    </mat-grid-tile>
                  </mat-grid-list>
                </div>
              </mat-expansion-panel>
              <p></p>
              <mat-expansion-panel>
                <mat-expansion-panel-header (click)="resetLollipop()">
                  <mat-panel-title>Gene Set Enrichment Analysis</mat-panel-title>
                  <mat-panel-description>
                    View GSEA results for the selected disease type.
                  </mat-panel-description>
                </mat-expansion-panel-header>
                <mat-expansion-panel class="info-card-small" expanded="false">
                  <mat-expansion-panel-header>
                    <mat-panel-title><mat-icon>info</mat-icon>What does this mean?</mat-panel-title>
                  </mat-expansion-panel-header>
                  <div class="info-text-small-div">
                    <p>Gene Set Enrichment Analysis (GSEA) identifies whether specific sets of genes related to biological functions are enriched in gene expression data.</p>
                    <p>The information from GSEA can help find biological differences between disease (sub-)types</p>
                  </div>
                </mat-expansion-panel>
                <div style="text-align: center;">
                  <app-gsea [input_condition_1]="'normal'" [input_condition_2]="'disease'"></app-gsea>
                </div>
              </mat-expansion-panel>


            </mat-accordion>
          </mat-expansion-panel>
          <p></p>
        </div>
        <!-- predict tab -->
        <div [hidden]="selectedTab.value!=tabs[1].value">
          <mat-card class="expression-upload-card">
            <mat-card-header>
              <div mat-card-avatar class="circle">1</div>
              <mat-card-title>Expression file upload</mat-card-title>
              <mat-card-subtitle>
                Upload gene/transcript expression for cancer type classification
              </mat-card-subtitle>
              <div class="example-expression-button-div">
                <button mat-button
                        class="example-expression-button"
                        (click)="flipExampleExpression()">
                  {{buttonText("expr")}}
                </button>
              </div>
            </mat-card-header>
            <mat-card-content>
              <ng-container *ngIf="showExpressionExample">
                <div class="example-expression-table-div">
                  <table mat-table [dataSource]="exampleExpressionData" class="mat-elevation-z8 example-expression-table">
                    <ng-container [matColumnDef]="column" *ngFor="let column of displayedCols">
                      <th mat-header-cell *matHeaderCellDef>{{ displayedColsValueMap.get(column) }}</th>
                      <td mat-cell *matCellDef="let emp">{{ emp[column] }}</td>
                    </ng-container>
                    <tr mat-header-row *matHeaderRowDef="displayedCols"></tr>
                    <tr mat-row *matRowDef="let row; columns: displayedCols"></tr>
                  </table>
                </div>
              </ng-container>
              <ngx-dropzone (change)="onExpressionUpload($event)"
                            multiple="false"
                            accept="{{acceptExpressionFiles()}}"
                            maxFileSize="{{maxFileSize}}"
                            disableClick="{{expressionUploaded()}}">
                <ngx-dropzone-label>
                  Drop your expression file here!
                  <img src="../../../assets/img/heat-map-svgrepo-com.png" width="15%" height="15%">
                  Or click the icon to select a file
                </ngx-dropzone-label>
                <ngx-dropzone-preview *ngFor="let f of uploadedExpressionFiles" style="text-align: center"
                                      [removable]="true" (removed)="onRemoveExpression(f)">
                  <ngx-dropzone-label>{{ f.name }}</ngx-dropzone-label>
                </ngx-dropzone-preview>
              </ngx-dropzone>
            </mat-card-content>
          </mat-card>
          <p></p>
          <mat-expansion-panel hideToggle>
            <mat-expansion-panel-header class="disable-ripple">
              <mat-panel-title>
                <div class="circle">2</div>&nbsp;Options
              </mat-panel-title>
              <mat-panel-description>
                spongEffects prediction parameters
              </mat-panel-description>
            </mat-expansion-panel-header>
            <hr class="my-4">
            <h6>Filtering thresholds:</h6>
            <mat-form-field>
              <mat-label>multiple miRNA sensitivity correlation (mscor)</mat-label>
              <input class="col-xs-3" type="number" [placeholder]="mscorDefault.toString()" matInput id="mscorSelect" [formControl]="mscorControl">
              <mat-error>Please provide a number between 0 and 10</mat-error>
            </mat-form-field>
            <mat-form-field>
              <mat-label>False discovery rate (FDR)</mat-label>
              <input class="col-xs-4 col-xs-offset-2" type="number" [placeholder]="fdrDefault.toString()" matInput id="fdrSelect" [formControl]="fdrControl">
              <mat-error>Please provide a number between 0 and 0.5</mat-error>
            </mat-form-field>
            <mat-form-field>
              <mat-label>Minimal expression value</mat-label>
              <input class="col-xs-4 col-xs-offset-2" type="number" [placeholder]="minExprDefault.toString()" matInput id="minExprSelect" [formControl]="minExprControl">
              <mat-error>Please provide a number between 0 and 1.000</mat-error>
            </mat-form-field>
            <p></p>
            <h6>Further settings:</h6>
            <mat-form-field>
              <mat-label>Enrichment method</mat-label>
              <mat-select [(value)]="methodDefault" [placeholder]="methodDefault">
                <mat-option *ngFor="let method of methods" [value]="method">{{method}}</mat-option>
              </mat-select>
            </mat-form-field>
            <mat-form-field>
              <mat-label>Minimum module size</mat-label>
              <input class="col-xs-3" matInput id="minSizeSelect" type="number" [placeholder]="minSizeDefault.toString()" [formControl]="minSizeControl">
              <mat-error>Please provide a number between 0 and 5.000</mat-error>
            </mat-form-field>
            <mat-form-field>
              <mat-label>Maximum module size</mat-label>
              <input class="col-xs-3" matInput id="maxSizeSelect" type="number" [placeholder]="maxSizeDefault.toString()" [formControl]="maxSizeControl">
              <mat-error>Please provide a number between 0 and 5.000</mat-error>
            </mat-form-field>
            <mat-checkbox class="check-box"
                          color="primary"
                          [(ngModel)]="logScaling">
              Apply log2 scaling factor to uploaded expression
            </mat-checkbox>
            <mat-checkbox class="check-box"
                          color="primary"
                          [(ngModel)]="predictSubtypes">
              Predict on subtype level (can lead to longer run times)
            </mat-checkbox>
          </mat-expansion-panel>
          <hr>
          <div class="run-button-div">
            <button mat-raised-button class="run-button-predict"
                    [disabled]="runButtonDisabled()"
                    (click)="predict()">
              Predict!
            </button>
            <mat-progress-bar class="predict-run-progress-bar"
                              [mode]="progressBarMode"
                              [value]="progressBarValue">
            </mat-progress-bar>
          </div>
          <div class="prediction-results">
            <div [hidden]="!(predictionLoading && predictionQueried)">
              Estimated run time: {{ estimatedRunTimeText()}}
            </div>
            <div class="predict-results" [hidden]="predictionLoading || !predictionQueried">
              <div class="predict-results-plots">
                <div class="type-prediction-div">
                  <div #typePredictPie class="type-prediction"></div>
                  <div class="type-legend">
                    <div class="legend">
                      <div class="legend-item">
                        <div class="percent-color-box">
                          <div class="percent-box-labels">
                          </div>
                        </div>
                        <div class="legend-labels">
                          <div class="child">1.0</div>
                          <div class="child">0.5</div>
                        </div>
                        <div class="legend-text">Overall class model accuracy</div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <p></p>
              <div class="type-prediction-text">
                Predicted type over all samples: {{this.predictedType}}
              </div>
            </div>
          </div>
        </div>
      </mat-sidenav-content>
    </mat-sidenav-container>
  </div>
</section>
